### PR TITLE
New branch 1

### PR DESCRIPTION
**Pull request by jtauber7**

### CODE REVIEW

File: Bulls_Cows.py [hunk 1]
```diff
@@ -66,7 +67,44 @@ def main():
...
+        print(f"| {str(i+1)+'.':^{c1}} | {row[0]:<{c2}} | {row[1]:^{c3}} | {str(row[2])+'s':^{c4}} |")
+
+
+def table_widths(stats_tab):
+    widths = [0] * len(stats_tab[0])
+    for row in stats_tab:
+        for i, word in enumerate(row):
+            if len(str(word)) > widths[i]:
+                widths[i] = len(str(word))
+    print(widths)
```
Comment (jtauber7) [inline Bulls_Cows.py:107]: aha..zapomenutý řádek..

File: Tic_Tac_Toe.py [hunk 2]
```diff
@@ -13,6 +13,9 @@ def main():
     print_board(board)
     board = [" " for i in range(0, 9)]
     while not evaulate_board(board,player):
+        if " " not in board:
+            print("The game ended with drew.")
```
Comment (Artimi) [inline Tic_Tac_Toe.py:17]: Typo `draw`

File: Bulls_Cows.py [hunk 3]
```diff
@@ -33,38 +35,76 @@ def result_print(result):
...
     end_time = time.time()
     time_to_guess = int(end_time - start_time)
-    end_game(attemp, time_to_guess)
+    end_game(attempt, time_to_guess, name)
+    stats_tab = user_stats(name, attempt, time_to_guess)
+    print_stats(stats_tab)
 
+def user_stats(name, attempt, time_to_guess):
+    with open("players_stats.txt", "a+") as file:
+        print(str(name)+",", str(attempt)+",", str(time_to_guess), file=file)
```
Comment (Artimi) [inline Bulls_Cows.py:76]: Isn't
```python
file.write(f'{name},{attempt},{time_to_guess}\n')
```
easier to read?

File: Bulls_Cows.py [hunk 4]
```diff
@@ -33,38 +35,76 @@ def result_print(result):
...
+        file.seek(0)
+        stats_tab = [i.strip("\n").split(", ") for i in file.readlines()]
+        for row in stats_tab:
+            row[1:] = (int(x) for x in row[1:])
+        return stats_tab
+
+def print_stats(stats_tab):
+    stats_tab.sort(key=lambda row: (row[1], row[2]))
```
Comment (Artimi) [inline Bulls_Cows.py:84]: :+1:

**Review by Artimi — COMMENTED** [review]
I only glanced over the code but it looks ok. I cannot merge it because it is your code so it is up to you to merge the pull request.

**Pull request merged**